app.js

```diff
@@ -0,0 +1,146 @@
+"use strict";
+
+(() => {
+  const API_BASE = 'https://restcountries.com/v3.1/all?fields=name,population,region,currencies'
+
+  const status = document.getElementById('status')
+  const output = document.getElementById('output')
+  const input = document.getElementById('q')
+  const form = document.getElementById('searchForm')
+  const btnAll = document.getElementById('btnAll')
+
+  const state = { countries: null }
+
+  const setBusy = b => output.setAttribute('aria-busy', String(b))
+  const showStatus = (msg, err = false) => {
+    status.classList.remove('d-none', 'alert-info', 'alert-danger')
+    if (!msg) {
+      status.classList.add('d-none')
+      status.textContent = ''
+      return
+    }
+    status.classList.add(err ? 'alert-danger' : 'alert-info')
+    status.textContent = msg
+  }
+
+  const number = n => Number(n || 0)
+  const fmt = n => number(n).toLocaleString('en-US')
+
+  const getData = url => fetch(url).then(r => r.json())
+
+  const loadAll = async () => {
+    if (Array.isArray(state.countries)) return state.countries
+    const data = await getData(`${API_BASE}/all?fields=name,population,region,currencies`)
+    state.countries = data
+    return data
+  }
+
+  const fetchAll = loadAll
+
+  const fetchByName = async name => {
```

Comment:
this is wrong, you were asked to use another endpoint on the server rather than to filter the `all` result